CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -1958,6 +1958,60 @@ + (NSArray*)getNotifications {
     return notifications;
 }
 
+#pragma mark - Retrieving notifications based on the flag remote/local
++ (NSArray*)getNotifications:(BOOL)combineWithApi {
+    if (combineWithApi) {
+        NSString *combinedURL = [self createDynamicURLForNotification];
+        NSMutableArray* combinedArray = [[self getNotifications] mutableCopy];
+        NSMutableArray *remoteNotifications = [self getRemoteNotifications:combinedURL];
```

Comment:
We need to merge the results without duplicates, only include `remoteNotifications` which are *not* included in `localNotifications`
Also make sure they are sorted correctly.

Reply:
When you build the array to return, please call `CPNotification initWithJson` for each entry

Reply:
We didn't use any custom object yet to pass the data, We have always passed the data by NSArray and to link with cross platform like flutter we will need a Native objects.

Reply:
@Azharhussain20  Can you please check tomorrow that Flutter still works with this new method? As we now use CPNotification.

Reply:
Ok sure.